raiden/transfer/channel.py

```diff
@@ -1561,11 +1562,12 @@ def handle_action_close(
 ) -> TransitionResult[NettingChannelState]:
     msg = "caller must make sure the ids match"
     assert channel_state.identifier == close.channel_identifier, msg
-
-    events = events_for_close(
-        channel_state=channel_state, block_number=block_number, block_hash=block_hash,
-        signed_close_tx=close.signed_close_tx
-    )
+    events = []
+    if close.participant1 == channel_state.our_state.address:
+        events = events_for_close(
+            channel_state=channel_state, block_number=block_number, block_hash=block_hash,
+            signed_close_tx=close.signed_close_tx
+        )
```

Comment:
can you please explain the reason behind this change?

why do we not care about `close.participant2`, for example?

Reply:
Now (since banche RLP-715) we have two channels per channel (one per side), but we only want to close it once, or else, it will fail. Also, the `close.signed_close_tx` will not verify for both sides of the channel.